List.cpp

```diff
@@ -0,0 +1,1038 @@
+            _tail = other._tail;
+        }
+        else if (pos == cbegin())
+        {
+            other._tail->_next = _head;
+            if (_head != nullptr)
+            {
+                _head->_prev = other._head;
+            }
+            _head = other._head;
+        }
+        else
+        {
+            iterator current{const_cast<Node<T> *>(pos.ptr())};
+            iterator fin(other._tail);
+
+            other.begin().ptr()->_prev = current.ptr()->_prev;
+            fin.ptr()->_next = current.ptr();
+
+            current.ptr()->_prev->_next = other.begin().ptr();
+            current.ptr()->_prev = fin.ptr();
+        }
+
+        _size += other._size;
+
+        other._size = 0;
+        other._head = other._tail = nullptr;
+    }
+
+    void reverse()
+    {
+        iterator it = begin();
+
+        while (it != end())
+        {
+            it.ptr()->rev();
+            --it;
+        }
+
+        _head->rev();
+        _tail->rev();
+        std::swap(_head, _tail);
+    }
+
+    void sort() { MergeSort(begin(), end(), _size); }
+
+    iterator begin() { return iterator(_head); }
+    const_iterator begin() const { return const_iterator(_head); }
+    const_iterator cbegin() const { return begin(); }
+
+    iterator end() { return iterator(_tail->_next); }
+    const_iterator end() const { return const_iterator(_tail->_next); }
+    const_iterator cend() const { return end(); }
+
+    riterator rbegin() { return riterator(_tail); }
+    const_riterator rbegin() const { return const_riterator(_tail); }
+    const_riterator crbegin() const { return rbegin(); }
+
+    riterator rend() { return riterator(_head->_prev); }
+    const_riterator rend() const { return const_riterator(_head->_prev); }
+    const_riterator crcend() const { return rend(); }
+};
+
+template <typename T>
+bool operator==(const Custom_List<T> &lhs, const std::list<T> &rhs)
+{
+    EXPECT_EQ(lhs.size(), rhs.size());
+    EXPECT_TRUE(std::equal(lhs.begin(), lhs.end(), rhs.begin()));
+    return true;
+}
+
+template <typename T>
+bool operator==(const Custom_List<T> &lhs, const Custom_List<T> &rhs)
+{
+    if (lhs.size() != rhs.size())
+    {
+        return false;
+    }
+    return std::equal(lhs.begin(), lhs.end(), rhs.begin());
+}
+
+TEST(List_Test, constructorByDefault)
+{
+    Custom_List<int> test;
+
+    EXPECT_TRUE(test.empty());
+}
+
+TEST(List_Test, constructorBySize)
+{
+    {
+        Custom_List<int> test(0);
+        EXPECT_TRUE(test.empty());
+    }
+
+    {
+        Custom_List<int> test(1);
+        EXPECT_EQ(test.size(), 1);
+    }
+
+    {
+        Custom_List<int> test(100);
+        EXPECT_EQ(test.size(), 100);
+    }
+
+    Custom_List<int> test1(8);
+    std::list<int> test2(8);
+
+    EXPECT_EQ(test1.size(), test2.size());
+}
+
+TEST(List_Test, constructorByInitList)
+{
+    {
+        std::initializer_list<int> tmp;
+        Custom_List<int> test(tmp);
+        EXPECT_EQ(test.size(), tmp.size());
```

Comment:
The tests must check not only the list size but also values and emptiness here and in other places.